clients/fj-svelte/src/lib/Client.svelte

```diff
@@ -33,80 +34,55 @@ with this program. If not, see <https://www.gnu.org/licenses/>.
   import Loading from "./Loading.svelte";
 
   import { get_username } from "../api";
-  import Countdown from "./Countdown.svelte";
+  import InlineCountdown from "./counters/InlineCountdown.svelte";
+  import PageCountdown from "./counters/PageCountdown.svelte";
+
+  // There's no cleaner way of doing this, right?
+  let destroyed = false;
+  onDestroy(() => {
+    destroyed = true;
+  });
 
   let username = "Loading...";
 
   get_username().then((name) => {
     username = name;
   });
 
-  let comp_times: CompTimes | undefined = undefined;
+  let compTimes: CompTimes | undefined = undefined;
   let questions: Record<string, QuestionMeta> | undefined = undefined;
-  let loading_errors: string[] = [];
+  let loadingErrors: string[] = [];
 
   get_comp_info().then((data) => {
     window.document.title = data.title;
   });
 
-  const load_questions = () => {
-    get_questions()
-      .then((q) => {
-        questions = q;
-      })
-      .catch((err) => {
-        loading_errors.push(err);
-        loading_errors = loading_errors;
-      });
-  };
-
-  let current_state: CompState | undefined = undefined;
-
-  let timer_ref = -1;
-
-  // Sets a timer to trigger a rerender to update the main content
-  const set_timer_for_next_state = () => {
-    if (comp_times == undefined) return;
-
-    current_state = get_current_comp_state(comp_times);
-
-    if (showing_questions_at_current_time(comp_times)) {
-      load_questions();
+  let timeStateData: TimeStateData | undefined = undefined;
+  runRepeatedly(() => {
+    if (compTimes !== undefined) {
+      timeStateData = getCurrentTimeStateData(compTimes);
     }
+  });
 
-    console.log(`changed current state to ${current_state}`);
-
-    timer_ref = setTimeout(
-      () => {
-        set_timer_for_next_state();
-      },
-      get_time_till_next_state(comp_times, current_state),
-    );
-  };
-
-  onDestroy(() => clearTimeout(timer_ref));
-
-  get_comp_times()
-    .then((t) => {
-      if (t === undefined) {
-        load_questions();
+  onMount(async () => {
+    try {
+      compTimes = await get_comp_times();
+      if (destroyed) return;
+      if (compTimes === undefined) {
+        loadingErrors.push("Could not load competition times");
         return;
       }
 
-      console.log("got times");
-
-      comp_times = t;
-
-      if (showing_questions_at_current_time(t)) {
-        load_questions();
-      }
+      // Wait until questions can be loaded
+      const millisecondsUntilQuestions = Math.max(0, compTimes.start.getTime() - Date.now()) + 500;
+      await new Promise((resolve) => setTimeout(resolve, millisecondsUntilQuestions));
+      if (destroyed) return;
 
-      set_timer_for_next_state();
-    })
-    .catch((err) => {
-      loading_errors.push(err);
-      loading_errors = loading_errors;
-    });
+      questions = await get_questions();
+    } catch (err: any) {
+      loadingErrors.push(err.message ?? err.toString());
```

Comment:
You need to do `loadingErrors = loadingErrors;` after this. There needs to be an actual assignment operator for a rerender to trigger.